worlds/shapez/regions.py

```diff
@@ -0,0 +1,272 @@
+from typing import Dict, Tuple, List
+
+from BaseClasses import Region, MultiWorld, LocationProgressType, ItemClassification, CollectionState
+from .items import ShapezItem
+from .locations import ShapezLocation
+from .data.strings import ITEMS, REGIONS, GOALS, LOCATIONS, OPTIONS
+from worlds.generic.Rules import add_rule
+
+shapesanity_processing = [REGIONS.full, REGIONS.half, REGIONS.piece, REGIONS.stitched, REGIONS.east_wind,
+                          REGIONS.half_half, REGIONS.col_east_wind, REGIONS.col_half_half, REGIONS.col_full,
+                          REGIONS.col_half]
+shapesanity_coloring = [REGIONS.uncol, REGIONS.painted, REGIONS.mixed]
+
+all_regions = [
+    REGIONS.menu, REGIONS.belt, REGIONS.extract, REGIONS.main,
+    REGIONS.levels_1, REGIONS.levels_2, REGIONS.levels_3, REGIONS.levels_4, REGIONS.levels_5,
+    REGIONS.upgrades_1, REGIONS.upgrades_2, REGIONS.upgrades_3, REGIONS.upgrades_4, REGIONS.upgrades_5,
+    REGIONS.paint_not_quad, REGIONS.cut_not_quad, REGIONS.rotate_cw, REGIONS.stack_shape, REGIONS.store_shape,
+    REGIONS.trash_shape, REGIONS.blueprint, REGIONS.wiring, REGIONS.mam, REGIONS.any_building,
+    REGIONS.all_buildings, REGIONS.all_buildings_x1_6_belt,
+    *[REGIONS.sanity(processing, coloring)
+      for processing in shapesanity_processing
+      for coloring in shapesanity_coloring],
+]
+
+
+def can_cut_half(state: CollectionState, player: int) -> bool:
+    return state.has(ITEMS.cutter, player)
+
+
+def can_rotate_90(state: CollectionState, player: int) -> bool:
+    return state.has_any([ITEMS.rotator, ITEMS.rotator_ccw], player)
```

Comment:
I've been advised that using tuples `(...)` instead of lists `[...]` or dicts `{...}` is preferable when calling `has_any` or `has_all`, though any iterable should work for those functions. Up to you.